internal/operands/common-templates/reconcile.go

```diff
@@ -0,0 +1,142 @@
+package common_templates
+
+import (
+	"fmt"
+	templatev1 "github.com/openshift/api/template/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sync"
+)
+
+var (
+	loadTemplatesOnce sync.Once
+	templatesBundle   []templatev1.Template
+)
+
+type commonTemplates struct{}
+
+func GetOperand() operands.Operand {
+	return &commonTemplates{}
+}
+
+func (c *commonTemplates) AddWatchTypesToScheme(s *runtime.Scheme) error {
+	return templatev1.Install(s)
+}
+
+func (c *commonTemplates) WatchClusterTypes() []runtime.Object {
+	return []runtime.Object{
+		&rbac.ClusterRole{},
+		&rbac.ClusterRoleBinding{},
+		&core.Namespace{},
+		&templatev1.Template{},
+	}
+}
+
+func (c *commonTemplates) WatchTypes() []runtime.Object {
+	return nil
+}
+
+func (c *commonTemplates) Reconcile(request *common.Request) error {
+	for _, f := range []func(*common.Request) error{
+		reconcileGoldenImagesNS,
+		reconcileViewRole,
+		reconcileViewRoleBinding,
+		reconcileEditRole,
+		reconcileTemplates,
+	} {
+		if err := f(request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *commonTemplates) Cleanup(request *common.Request) error {
+	objects := []controllerutil.Object{
+		newGoldenImagesNS(GoldenImagesNSname),
+		newViewRole(GoldenImagesNSname),
+		newViewRoleBinding(GoldenImagesNSname),
+		newEditRole(GoldenImagesNSname),
+	}
+	for _, template := range templatesBundle {
+		objects = append(objects, &template)
```

Comment:
Namespace needs to be set for these templates. Otherwise the `Delete` call below will not know from which namespace the template should be removed.